### ISSUE TITLE
Segfault with basic kernel from AMDGPU.jl doc on LUMI

### ISSUE THREAD

**Issue by omlins** [JuliaGPU/AMDGPU.jl#308]
A segfault is obtained with this basic kernel from AMDGPU.jl doc on LUMI:
```julia
N = 32
a = rand(Float64, N);
b = rand(Float64, N);
c_cpu = a + b;
using AMDGPU
a_d = ROCArray(a);
b_d = ROCArray(b);
c_d = similar(a_d);

function vadd!(c, a, b)
    i = workitemIdx().x
    c[i] = a[i] + b[i]
    return
end

@roc groupsize=N vadd!(c_d, a_d, b_d)
```
Segfault:
```
julia> N = 32
32

julia> a = rand(Float64, N);

julia> b = rand(Float64, N);

julia> c_cpu = a + b;


julia> using AMDGPU

julia> a_d = ROCArray(a);

julia> b_d = ROCArray(b);

julia> c_d = similar(a_d);

julia> function vadd!(c, a, b)
           i = workitemIdx().x
           c[i] = a[i] + b[i]
           return
       end
vadd! (generic function with 1 method)

julia> @roc groupsize=N vadd!(c_d, a_d, b_d)

signal (11): Segmentation fault
in expression starting at REPL[10]:1
_ZN4llvm15ScalarEvolution10createSCEVEPNS_5ValueE at /pfs/lustrep2/scratch/project_465000105/saomlin/julia_local/julia-1.8.2/bin/../lib/julia/libLLVM-13jl.so (unknown line)
unknown function (ip: 0x32bc96f)
Allocations: 34856859 (Pool: 34841656; Big: 15203); GC: 33
Segmentation fault
saomlin@nid005002:~> 
```

Packages installed:
```
(v1.8) pkg> status
Status `/pfs/lustrep2/scratch/project_465000105/saomlin/julia_local/julia_depot/environments/v1.8/Project.toml`
  [21141c5a] AMDGPU v0.4.3 `/scratch/project_465000105/saomlin/julia_local/julia_depot/dev/AMDGPU`
  [052768ef] CUDA v3.12.0
  [d35fcfd7] CellArrays v0.1.1
  [94395366] ParallelStencil v0.6.0 `/scratch/project_465000105/saomlin/julia_local/julia_depot/dev/ParallelStencil`
  [90137ffa] StaticArrays v1.5.9
```

AMDGPU versioninfo:
```
julia> AMDGPU.versioninfo()
HSA Runtime (ready)
- Path: /opt/rocm/lib/libhsa-runtime64.so.1
- Version: 1.1.0
ld.lld (ready)
- Path: /opt/rocm/llvm/bin/ld.lld
ROCm-Device-Libs (ready)
- Path: /opt/rocm/amdgcn/bitcode
HIP Runtime (ready)
- Path: /opt/rocm/lib/libamdhip64.so.5
rocBLAS (ready)
- Path: /opt/rocm/lib/librocblas.so
rocSOLVER (ready)
- Path: /opt/rocm/lib/librocsolver.so
rocALUTION (ready)
- Path: /opt/rocm/lib/librocalution.so
rocSPARSE (ready)
- Path: /opt/rocm/lib/librocsparse.so
rocRAND (ready)
- Path: /opt/rocm/lib/librocrand.so
rocFFT (ready)
- Path: /opt/rocm/lib/librocfft.so
MIOpen (MISSING)
HSA Agents (5):
- CPU-XX [AMD EPYC 7A53 64-Core Processor]
- CPU-XX [AMD EPYC 7A53 64-Core Processor]
- CPU-XX [AMD EPYC 7A53 64-Core Processor]
- CPU-XX [AMD EPYC 7A53 64-Core Processor]
- GPU-ae2663b2e5711fdb [gfx90a]
```

**Comment by pxl-th** [comment]
It looks similar to what I had on 1.8 as well and I'd guess that this is due to alloca issue that was [fixed](https://github.com/JuliaLang/julia/pull/45900) on 1.9 nightly version.

If you can try 1.9, it should work:

```julia
pxl-th@Yotun:~/.julia/dev/AMDGPU$ amdjl 
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.9.0-DEV.1437 (2022-09-26)
 _/ |\__'_|_|_|\__'_|  |  Commit 26304f763cf (15 days old master)
|__/                   |

julia> using AMDGPU

julia> N = 32;

julia> a = rand(Float64, N);

julia> b = rand(Float64, N);

julia> c_cpu = a + b;

julia> a_d = ROCArray(a);

julia> b_d = ROCArray(b);

julia> c_d = similar(a_d);

julia> function vadd!(c, a, b)
           i = workitemIdx().x
           c[i] = a[i] + b[i]
           return
       end
vadd! (generic function with 1 method)

julia> wait(@roc groupsize=N vadd!(c_d, a_d, b_d));

julia> Array(c_d) ≈ c_cpu
true
```

**Comment by jpsamaroo** [comment]
Yes, it's the alloca issue which was fixed on 1.9. It's unfortunately not backportable to 1.8, so there's not much I can do.

**Comment by omlins** [comment]
> Yes, it's the alloca issue which was fixed on 1.9. It's unfortunately not backportable to 1.8, so there's not much I can do.

Is there any setup using Julia 1.8 where AMDGPU.jl works?
